plotting/plot_handler.py

```diff
@@ -0,0 +1,107 @@
+    :param instrument_name: The name of the beamline/spectrometer/instrument.
+    :param rb_number: The ISIS RB number.
+    :param run_number: The run number on the given instrument for the given RB number.
+    :param plot_type:
+        The type of plot file / file extension to be searched for (e.g. "png"). Can be a list of extensions.
+        If None, a list of common graphics file extensions is searched for.
+    """
+
+    def __init__(self, instrument_name, rb_number, run_number, plot_type=None):
+        self.instrument_name = instrument_name
+        self.rb_number = rb_number
+        self.run_number = run_number
+        if plot_type is None:
+            # list of plot types the handler looks for if none is provided; it searches for the types in the order of
+            # the list and selects the first one that exists. In future: could add vector graphics (.pdf, .eps)
+            self.plot_type = ["png", "jpg", "bmp", "gif", "tiff"]
+        elif isinstance(plot_type, str):
+            # if only a single str is passed for a plot type then pass it into a list (so that a single for-loop can be
+            # used to loop over plot types).
+            self.plot_type = [plot_type]
+        else:
+            self.plot_type = plot_type
+        # dictionary of regular expressions for each instrument to describe the possible prefixes in the file name(s)
+        self._instrument_dict = {"MARI": "MAR[I]", "WISH": "WISH"}
+        # path to directory with files for given instrument and RB number
+        self._rbfolder = "/instrument/" + self.instrument_name + "/RBNumber/RB" + str(
+            self.rb_number) + "/" + str(self.run_number) + "/autoreduced/"
+        # parameter to store the file names of any existing plot files found
+        self._existing_plot_files = []
+
+    def _regexp_for_file_name(self, plot_type):
+        """
+        Regular expression used for looking for plot files. It assumes that the file names follow the convention
+        <instrument_abbreviation><run_number>*<.png or other extension>
+        """
+        try:
+            _regexp = self._instrument_dict[self.instrument_name] + str(
+                self.run_number) + "*." + plot_type
+            return _regexp
+        except KeyError:  # if the instrument name does not appear in the dictionary of known instruments
+            LOGGER.info("The instrument name is not recognised")
+
+    def _check_for_plot_files(self):
+        """
+        Searches the CEPH directory for existing plot files using the directory specified by instrument_name, rb_number,
+        run_number, and a regular expression for the file name based on the instrument_name, run_number and plot_type.
+        """
+        # start sftpclient
+        client = SFTPClient()
+        # initialise list to store names of existing files matching the search
+        _found_files = []
+        for plot_type_i in self.plot_type:
```

Comment:
I would change `plot_type_i` to something like `plot_type` or simply `type`.
Just because this value will be the item within the list, rather than an index.

Reply:
use plot_type rather than type. `type` is a reserved key word in python